src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php

```diff
@@ -25,7 +25,7 @@ private function filterFilesByExtension(Traversable $files, string $fileExtensio
 
         /* @var $file \SplFileInfo */
         foreach ($files as $file) {
-            if (! preg_match($extensionMatcher, $file->getBasename())) {
+            if ($file->isDir() || ! preg_match($extensionMatcher, $file->getBasename())) {
```

Comment:
Only functional changes please - performance only where performance improvement is needed.

This exposed another bug by the way: creating a directory or a symlink `foo.php` will likely need a test :-\